esmvaltool/diag_scripts/portrait_plot.py

```diff
@@ -0,0 +1,505 @@
+"""Portrait Plot Diagnostic.
+
+Description
+-----------
+This diagnostic provides plot functionalities for performance metrics,
+and is written to be as flexible as possible to be adaptable to further use
+cases. X and Y axis, grouping parameter and splits for each rectangle can be
+configured in the recipe. All `*_by` parameters can be set to any metadata
+key. To split by 'reference' this key needs to be set as extra_facet in recipe.
+
+Authors
+-------
+- Lukas Lindenlaub (Universität Bremen, Germany)
+- Diego Cammarano
+
+Configuration parameters through recipe:
+----------------------------------------
+normalize: str or None, optional
+    ('mean', 'median', 'centered_mean', 'centered_median', None).
+    Divide by median or mean if not None. Subtract median/mean afterwards if
+    centered.
+    By default 'centered_median'.
+x_by: str, optional
+    Metadata key for x coordinate.
+    By default 'alias'.
+y_by: str, optional
+    Metadata key for y coordinate.
+    By default 'variable_group'.
+group_by: str, optional
+    Metadata key for grouping.
+    Grouping is always applied in x direction. Can be set to None to skip
+    grouping into subplots.
+    By default 'project'.
+split_by: str, optional
+    The rectangles can be split into 2-4 triangles. This is used to show
+    metrics for different references. For this case there is no need to change
+    this parameter. Multiple variables can be set in the recipe with `split`
```

Comment:
```suggestion
    this parameter. Multiple variables can be set in the recipe with ``split``
```

Single backticks are rendered italic, to get monospace font you need double backticks. Applies not only to this line.